tests/test_render.py

```diff
@@ -45,6 +45,21 @@ def wait_for_text_to_equal(self, selector, assertion_text):
 
         raise exception
 
+    def wait_for_text_to_not_equal(self, selector, assertion_text):
+        start_time = time.time()
+        exception = Exception('Time ran out, {} on {} not found'.format(
+            assertion_text, selector))
+        while time.time() < start_time + 20:
```

Comment:
There are bulit-in wait-for tools provided by selenium that we should be using instead of doing this timing logic ourselves: https://selenium-python.readthedocs.io/waits.html

Reply:
Good point, although I just copied the function from above - perhaps we can make a separate PR that fixes up these integration tests with the proper Selenium api's etc.

Reply:
Sure, but I often find that if we do it in a separate PR, it frequently doesn't get done. I'd much rather do things the Right Way the first time.

Reply:
Since I just copy-pasted the function above, I feel asking me to refactor these functions is a bit out of scope - which is why I suggest doing it in another PR, or creating an issue for this. I'd rather see this PR get through a bit quicker than getting bogged down with - although good and reasonable - suggestions to fix other things. Technically, the Right Way could've been done when the function I copy-pasted was first added.

Reply:
#103 here's an issue for it.